File: leetcode/stackAndQueue/2000.cpp
#include <algorithm>
#include <functional> //包含std::greater 和 std::less
#include <iostream>
#include <limits.h>
#include <math.h>
#include <queue>
#include <stack>
#include <unordered_set>
#include <vector>

using namespace std;
struct TreeNode {
  int val;
  TreeNode *left;
  TreeNode *right;
  TreeNode() : val(0), left(nullptr), right(nullptr) {}
  TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
  TreeNode(int x, TreeNode *left, TreeNode *right)
      : val(x), left(left), right(right) {}
};
struct ListNode {
  int val;
  ListNode *next;
  ListNode() : val(0), next(nullptr) {}
  ListNode(int x) : val(x), next(nullptr) {}
  ListNode(int x, ListNode *next) : val(x), next(next) {}
};
class Solution {
public:
  string reversePrefix(string word, char ch) {
    stack<char> st;
    string result;
    int n = word.size();
    int i = 0;
    bool meet = false;
    for (; i < n; i++) {
      if (ch != word[i]) {
        st.push(word[i]);
      } else {
        st.push(word[i]);
        i++;
        meet = true;
        break;
      }
    }
    if(meet){
      while (!st.empty()) {
        result.push_back(st.top());
        st.pop();
      }
      for (; i < n; i++) {
        result.push_back(word[i]);
      }
      return result;
    }else{
      return word;
    }
    return word;
  }
};
int main() {
  // 示例二叉树
  Solution solution;
  string word = "abcd";
  char ch = 'z';
  string result = solution.reversePrefix(word, ch);
  std::cout << "result: " << result << std::endl;
}

File: leetcode/stackAndQueue/2000.py
from typing import List
from queue import Queue


class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        st: list = []
        result: List[str] = []
        n: int = len(word)
        i: int = 0
        meet: bool = False
        for i in range(n):
            if ch != word[i]:
                st.append(word[i])
            else:
                st.append(word[i])
                i += 1
                meet = True
                break
        if meet:
            while not 0 == len(st):
                result.append(st[-1])
                st.pop()
            for m in range(i, n):
                result.append(word[m])
            return "".join(result)
        else:
            return word
        return word


if __name__ == "__main__":
    solution = Solution()
    word: str = "abcdefd"
    ch: str = "d"
    result = solution.reversePrefix(word, ch)
    print(f"result = {result}")
